terminal.go

```diff
@@ -10,13 +11,67 @@ import (
 type Terminal struct {
 	fd       int
 	oldState *term.State
+	term     *term.Terminal
+	Out      io.Writer
+}
```

Comment:
```suggestion
 type Terminal struct {
	*term.Terminal
 	fd       int
 	oldState *term.State
	Out      io.Writer
}
```
Maybe this, would avoid to define Readline and SetPrompt

Reply:
I'm trying to control the exact exposed surface vs having users know both packages/types

but maybe... tbd